core/dbt/compilation.py

```diff
@@ -195,25 +195,35 @@ def compile(self, manifest):
                 manifest.macros.items()):
             stats[node.resource_type] += 1
 
-        self.write_graph_file(linker, manifest)
+        if write:
+            self.write_graph_file(linker, manifest)
         print_compile_stats(stats)
 
         return linker
 
 
-def compile_manifest(config, manifest):
+def compile_manifest(config, manifest, write=True):
     compiler = Compiler(config)
     compiler.initialize()
-    return compiler.compile(manifest)
+    return compiler.compile(manifest, write=write)
 
 
-def compile_node(adapter, config, node, manifest, extra_context):
+def _is_writable(node):
+    if not node.injected_sql:
+        return False
+
+    if dbt.utils.is_type(node, NodeType.Archive):
```

Comment:
will this manifest with us not writing compiled analyses to the target directory? If i'm reading this right, i think the behavior specified here would be undesirable, as the compiled sql in the target dir is the primary way that folks interact with analyses

Reply:
I'm afraid I don't understand - this is the same logic as the old code it replaced here: https://github.com/fishtown-analytics/dbt/blob/dev/wilt-chamberlain/core/dbt/compilation.py#L215-L216 except split out into a function for readability.

Reply:
roger, makes sense! We compile analyses, but they don't end up in the `target/run/` dir because they aren't run